gemstone/src/swapper/mayan/mayan_swift.rs

```diff
@@ -0,0 +1,412 @@
+        &self,
+        token_in: &str,
+        amount_in: U256,
+        params: OrderParams,
+        submission_fee: U256,
+        signed_order_hash: Vec<u8>,
+        permit_params: PermitParams,
+    ) -> Result<Vec<u8>, MayanSwiftError> {
+        let call_data = IMayanSwift::createOrderWithSigCall {
+            tokenIn: Address::from_str(token_in).map_err(|e| MayanSwiftError::ABIError {
+                msg: format!("Invalid token address: {}", e),
+            })?,
+            amountIn: amount_in,
+            params: self.convert_order_params(params),
+            submissionFee: submission_fee,
+            signedOrderHash: signed_order_hash.into(),
+            permitParams: self.convert_permit_params(permit_params),
+        }
+        .abi_encode();
+
+        Ok(call_data)
+    }
+
+    pub async fn create_order_with_sig(
+        &self,
+        from: &str,
+        token_in: &str,
+        amount_in: &str,
+        params: OrderParams,
+        submission_fee: &str,
+        signed_order_hash: Vec<u8>,
+        permit_params: PermitParams,
+    ) -> Result<String, MayanSwiftError> {
+        let call_data = self
+            .encode_create_order_with_sig(
+                token_in,
+                U256::from_str(amount_in).map_err(|_| MayanSwiftError::InvalidAmount)?,
+                params,
+                U256::from_str(submission_fee).map_err(|_| MayanSwiftError::InvalidAmount)?,
+                signed_order_hash,
+                permit_params,
+            )
+            .await?;
+
+        let create_order_call = EthereumRpc::Call(TransactionObject::new_call_with_from(from, &self.address, call_data), BlockParameter::Latest);
+
+        let response = jsonrpc_call(&create_order_call, self.provider.clone(), &self.chain)
+            .await
+            .map_err(|e| MayanSwiftError::CallFailed { msg: e.to_string() })?;
+
+        let result: String = response.extract_result().map_err(|e| MayanSwiftError::InvalidResponse { msg: e.to_string() })?;
+
+        Ok(result)
+    }
+
+    pub fn convert_permit_params(&self, permit_params: PermitParams) -> IMayanSwift::PermitParams {
+        IMayanSwift::PermitParams {
+            value: U256::from_str(&permit_params.value).unwrap(),
+            deadline: U256::from(permit_params.deadline),
+            v: permit_params.v.into(),
+            r: permit_params.r.into(),
+            s: permit_params.s.into(),
+        }
+    }
+
+    // Helper method to convert our native OrderParams to contract format
+    pub fn convert_order_params(&self, params: OrderParams) -> IMayanSwift::OrderParams {
+        params.to_contract_params()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{future::pending, time::Duration};
+
+    use async_std::future::timeout;
+    use async_trait::async_trait;
+
+    use crate::network::{mock::AlienProviderWarp, AlienError, AlienTarget, Data};
+
+    #[derive(Debug)]
+    pub struct AlienProviderMock {
```

Comment:
Because you move `AlienProviderMock` out of `cfg(test)` so you can reuse it from `network::mock`

Reply:
true, but I removed it